ext/LuxorTensorPlot/src/viz_contraction.jl

```diff
@@ -0,0 +1,138 @@
+function OMEinsumContractionOrders.ein2elimination(ein::NestedEinsum{T}) where{T}
+    elimination_order = Vector{T}()
+    _ein2elimination!(ein, elimination_order)
+    return elimination_order
+end
+
+function OMEinsumContractionOrders.ein2elimination(ein::SlicedEinsum{T, NestedEinsum{T}}) where{T}
+    elimination_order = Vector{T}()
+    _ein2elimination!(ein.eins, elimination_order)
+    # the slicing indices are eliminated at the end
+    return vcat(elimination_order, ein.slicing)
+end
+
+function _ein2elimination!(ein::NestedEinsum{T}, elimination_order::Vector{T}) where{T}
+    if ein.tensorindex == -1
+        for arg in ein.args
+            _ein2elimination!(arg, elimination_order)
+        end
+        iy = unique(vcat(getiyv(ein.eins)...))
+        for ix in unique(vcat(getixsv(ein.eins)...))
+            if !(ix in iy) && !(ix in elimination_order)
+                push!(elimination_order, ix)
+            end
+        end
+    end
+    return elimination_order
+end
+
+function elimination_frame(GViz, tng::TensorNetworkGraph{TG, TL}, elimination_order::Vector{TL}, i::Int; filename = nothing, color = (0.5, 0.5, 0.5, 0.5)) where{TG, TL}
+    GViz2 = deepcopy(GViz)
+    for j in 1:i
+        id = _get_key(tng.indices_labels, elimination_order[j])
+        GViz2.vertex_colors[id] = color
+    end
+    return show_graph(GViz2, filename = filename)
+end
+
+function OMEinsumContractionOrders.viz_contraction(ein::T, args...; kwargs...) where{T <: AbstractEinsum}
+    throw(ArgumentError("Only NestedEinsum and SlicedEinsum{T, NestedEinsum{T}} have contraction order"))
+end
+
+"""
+    viz_contraction(ein::ET; locs=StressLayout(), framerate=30, filename="contraction", pathname=".", create_gif=false, create_video=true, color=(0.5, 0.5, 0.5, 0.5), show_progress=false) where {ET <: Union{NestedEinsum, SlicedEinsum}}
+
+Visualize the contraction process of a tensor network.
+
+# Arguments
+- `ein::ET`: The tensor network to visualize.
+- `locs`: The layout algorithm to use for positioning the nodes in the graph. Default is `StressLayout()`.
+- `framerate`: The frame rate of the animation. Default is 30.
```

Comment:
Default framerate is too high.

Reply:
I lowered the default `framerate` to `10`, may be we can use default time length of the video instead of frame rate since size of different tensor network can be quite different.